src/batch.js

```diff
@@ -1,28 +1,221 @@
 'use strict'
 
+const router = require('lambda-router/src/router')
+const { expectation } = require('sinon')
+
 const { HttpError } = require('./httpError')
 
 const batchDefaultConfig = {
-  maxBatchSize: 20
+  maxBatchSize: 20,
+  handler: undefined
 }
 
 module.exports = {
   batchHandler
 }
 
-async function batchHandler({ routes, config, extractPathParameters }, event, context) {
+async function batchHandler({ route, config }, event, context) {
   config = { ...batchDefaultConfig, ...config }
+  const body = event.body
 
   // Validate
-  // validate json body has been parsed
-  // validate body.requests has array of requests
-  // validate each body.requests[]
+  validateBatchRequest(body, config.maxBatchSize)
 
   // build graph of requests
+  const orderedRequests = tSort(body.requests)
+  const { asyncRequests, syncRequests } = splitAsyncRequests(orderedRequests)
```

Comment:
What is this? I don't see anything in the JSON batch spec about sync vs async requests. Shouldn't they all be async?

Reply:
Sync requests are requests that must be executed one after the other in the case of dependsOn.

So:
{id:"A"}
{id:"B", dependsOn:["A"]}
{Id:"C"}

A and B will be categorized as syncRequests and executed in the order: A, B.
C will be categorized as an asyncRequest and be executed alongside A&B.

Reply:
Per slack convo:

Given
```
{id:"A"}
{id:"B", dependsOn:["A"]}
{id:"C"}
{id:"D", dependsOn:["C, B"]}
{id:"E", dependsOn:["C"]}
```
It should be possible for E to execute before D if B takes too long.

Reply:
Updated, ended up changing flow around to not build graph but simply execute as it steps through the dependency tree. Should accomplish this and is quite a bit simpler.

Reply:
Do we still have the ability to detect invalid trees and throw before we starting routing with that solution?

I don't want to execute 3 batch requests then figure out the last one has a circular dependency.

Reply:
We do, but it's now done as the last step in validation instead of during the ordered graph.